### PR TITLE
[XC] Fix support for x:TypeArguments on root node

### PR DESCRIPTION
**Pull request by simonrozsival**
### Description of Change

We weren't initializing the `XmlType` of the root node properly and this PR fixes it:

https://github.com/dotnet/maui/blob/1594fd2e5960e21a6a23ab2d43f3f4e7556943b4/src/Controls/src/Build.Tasks/XamlTask.cs#L59-L61

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/25172

### CODE REVIEW

File: src/Controls/tests/Xaml.UnitTests/Issues/Maui25172.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Controls.Core.UnitTests;
+using Microsoft.Maui.Dispatching;
+using Microsoft.Maui.UnitTests;
+using NUnit.Framework;
+
+namespace Microsoft.Maui.Controls.Xaml.UnitTests;
+
+[XamlCompilation(XamlCompilationOptions.Skip)]
```
Comment (StephaneDelcroix) [inline src/Controls/tests/Xaml.UnitTests/Issues/Maui25172.xaml.cs:9]: remove this, and also test that the instantiated type inherits from the right type

**Review by StephaneDelcroix — CHANGES_REQUESTED** [review]

File: src/Controls/tests/Xaml.UnitTests/Issues/Maui25172.xaml.cs (hunk 1 above)
Comment (simonrozsival) [reply src/Controls/tests/Xaml.UnitTests/Issues/Maui25172.xaml.cs]: fixed

**Review by StephaneDelcroix — APPROVED** [review]

**Comment by bcaceiro** [comment]
Should this be available in latest nightly for .net8? Because I started getting the same issue in .net 8 latest nightlie

**Comment by simonrozsival** [comment]
> Should this be available in latest nightly for .net8? Because I started getting the same issue in .net 8 latest nightlie

@bcaceiro this fix was targeting only .NET 9. Please open a new issue if there has been a regression in the .NET 8 nightly build. Thanks!

**Pull request merged**